contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,462 @@
+      _target,
+      _tokensToDistribute), "Unable to transfer tokens");
+
+    emit TokensCollected(
+      _target,
+      _vestingScheduleIndex,
+      _vestingSchedule.vestingId,
+      _tokensToDistribute
+    );
+  }
+
+  /**
+  * @notice Initiate a vesting schedule for an employee or affiliate
+  * @param _target Address of the employee or the affiliate
+  * @param _totalAllocation Total number of tokens allocated for the target
+  * @param _vestingDuration Total duration of the vesting schedule
+  * @param _startDate Start date of the vesting schedule
+  * @param _vestingFrequency Frequency of release of tokens
+  */
+  function initiateVestingScheduleIterate(
+    address _target,
+    uint256 _totalAllocation,
+    uint256 _vestingDuration,
+    uint256 _startDate,
+    uint256 _vestingFrequency
+  )
+    public
+    withPerm(ISSUER)
+  {
+    require(_target != address(0), "The target should be a valid addrss");
+    require(_totalAllocation != 0, "The total allocation should not be 0");
+    require(_vestingDuration != 0, "The vestingDuration should not be 0");
+    require(_startDate >= now, "The starting date should be after now");
+    require(
+      _vestingFrequency != 0 && _vestingFrequency <= _vestingDuration,
+      "The vestingFrequency should not be 0 and it should be less than the duration");
+    require(_vestingDuration % _vestingFrequency == 0, "The vesting frequency should be a multiple of the vesting duration");
+
+    uint256 _numTranches = _vestingDuration.div(_vestingFrequency);
+    uint256 _tokensPerTranche = _totalAllocation.div(_numTranches);
+
+    bytes32 _vestingId = keccak256(
+      abi.encodePacked(
+        block.timestamp,
+        _target,
+        _totalAllocation,
+        _vestingDuration,
+        _startDate,
+        _vestingFrequency
+      )
+    );
+
+    uint256 _vestingScheduleIndex = vestingScheduleIndex[_target];
+    vestingScheduleIndex[_target] += 1;
+
+    uint256 _numUnvestedTokens = _totalAllocation;
+
+    userToVestingSchedules[_target] = VestingSchedule({
+      vestingId: _vestingId,
+      totalAllocation: _totalAllocation,
+      vestingDuration: _vestingDuration,
+      startDate: _startDate,
+      vestingFrequency: _vestingFrequency,
+      numVestedTokens: 0,
+      numUnvestedTokens: _numUnvestedTokens,
+      numClaimedVestedTokens: 0,
+      numUnclaimedVestedTokens: 0,
+      tokensPerTranche: _tokensPerTranche
+    });
+
+    uint256 _tokensToSend = _calculateTokensToSend(_totalAllocation);
+
+    // Remove tokens from the excess token count if existing tokens are used
+    if (_tokensToSend != _totalAllocation) {
+      numExcessTokens = numExcessTokens.sub(_totalAllocation.sub(_tokensToSend));
+    }
+
+    if (_tokensToSend != 0) {
+      require(ISecurityToken(securityToken).transferFrom(
+        msg.sender,
+        address(this),
+        _tokensToSend), "Unable to transfer tokens");
+    }
+
+    emit VestingStarted(
```

Comment:
emit the `vestingScheduleIndex` so the dApp know that for the _vestee there is a new vesting created with its index is `_vestingScheduleIndex`.